File: 141_linked_list_cycle/141_linked_list_cycle.py
class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        # slow = head
        # fast = head

        # while fast and fast.next:
        #     slow = slow.next
        #     fast = fast.next.next
        #     if slow == fast:
        #         return True
        # return False

        res = {}
        current = head
        while current:
            if current in res:
                return True
            else: res[current] = True
            current = current.next
        return False

File: 141_linked_list_cycle/141_linked_list_cycle.ts
function hasCycle(head: ListNode | null): boolean {
  let current: ListNode | null = head;
  const res = new Set<ListNode>();

  while (current !== null) {
    if (res.has(current)) {
      return true; // Cycle detected
    } else {
      res.add(current);
      current = current.next;
    }
  }

  return false; // No cycle found
}
